src/index.ts

```diff
@@ -194,4 +195,67 @@ function countTokens(encoding: Tiktoken, text: string | undefined): number {
   return encoding.encode(text).length;
 }
 
+async function countImageTokens(contentPart: OpenAI.Chat.ChatCompletionContentPartImage, chatModel: string): Promise<number> {
+  const visionRatio = (chatModel !== 'gpt-4o-mini') ? 1 : gpt4oMiniRatio;
+
+  let tokens = fixedPrice;
+  
+  const detail = contentPart.image_url.detail;
```

Comment:
Consider treating incoming data as a black box. The 'image_url' property might be nullable, and 'detail' could be undefined. Add null checks or use the optional chaining operator (?.) to handle potential undefined values safely